controller/controlplane/controller_reconciler_utils.go

```diff
@@ -467,6 +468,24 @@ func (r *Reconciler) ensureAdmissionWebhookCertificateSecret(
 	matchingLabels := client.MatchingLabels{
 		consts.SecretUsedByServiceLabel: consts.ControlPlaneServiceKindWebhook,
 	}
+	if !isAdmissionWebhookEnabled(ctx, r.Client, logger, cp) {
```

Comment:
I guess this call can be considered quite cheap (given it fetches the pod container from the local cache), but we might calculate it once in `ensureWebhookResources` and pass it along to all functions depending on that. The drawback I can see is the growing number of parameters to be passed. Have you considered it?

Reply:
I've considered it and had exactly the same thoughts about that.

Let me try to pass that as an arg b4bec069e94a33b3e903eda5018e689f448fa683

Reply:
> Let me try to pass that as an arg https://github.com/Kong/gateway-operator/commit/b4bec069e94a33b3e903eda5018e689f448fa683

Not great, not terrible. I think calling `isAdmissionWebhookEnabled` for readability's and maintainability's sake might make sense. Let's leave it - we'll always have something to improve in terms of performance, and I don't think this will be the most burning issue. 🙂